scripts/isCI.js

```diff
@@ -0,0 +1,21 @@
+// Logic and values based on is-ci https://github.com/watson/is-ci/blob/master/bin.js
+// which uses ci-info under the hood https://github.com/watson/ci-info/blob/master/index.js
+// This just extracts the specific parts we need without adding more dependencies
+
+const { env } = process;
+
+const isCI = !!(
+  env.CI !== "false" && // Bypass all checks if CI env is explicitly set to 'false'
```

Comment:
While I do appreciate the robust approach, why did we decide to cover all possible types of CI providers? Just remember that we'd need to maintain this script going forward, and I could see it becoming out of date quickly.